airflow/operators/dagrun_operator.py

```diff
@@ -75,24 +63,21 @@ def __init__(
             self.execution_date = None
         else:
             raise TypeError(
-                'Expected str or datetime.datetime type '
-                'for execution_date. Got {}'.format(
-                    type(execution_date)))
+                "Expected str or datetime.datetime type for execution_date. "
+                "Got {}".format(type(execution_date))
+            )
 
     def execute(self, context):
```

Comment:
Opportunity for type annotation :-)

Reply:
The `__init__` arguments are already type annotated. Personally not a fan of these `if/else` type checks everywhere, it's against Python's nature :-)  Do you think we should remove it?

Reply:
No, it is okay like this.